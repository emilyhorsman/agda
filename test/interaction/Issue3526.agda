open import Agda.Builtin.Bool
open import Agda.Builtin.Equality

idTrue : ∀ b → b ≡ true → Bool
idTrue b eq = {!!}

-- C-c C-c b RET gives us:
-- idTrue false eq = {!!}
-- idTrue true eq = {!!}

-- Now that we can leave `idTrue false eq` out because it's
-- trivially an impossible clause, it'd be nice to only get:
-- idTrue true eq = {!!}


-- But we should still get an absurd clause when we explictly
-- ask for it as in the following example:
-- C-c C-c e RET should give use
-- absurd ()

data Empty : Set where

absurd : ∀ {a} {A : Set a} → Empty → A
absurd e = {!!}

-- Similarly, C-C C-c b eq RET should give us
-- idTrue' false ()
-- idTrue' true refl = {!!}

idTrue' : ∀ b → b ≡ true → Bool
idTrue' b eq = {!!}
