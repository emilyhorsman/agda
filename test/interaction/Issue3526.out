(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : Bool ?1 : A ?2 : Bool " nil)
((last . 1) . (agda2-goals-action '(0 1 2)))
((last . 2) . (agda2-make-case-action '("idTrue true eq = ?")))
((last . 1) . (agda2-goals-action '(0 1 2)))
((last . 2) . (agda2-make-case-action '("absurd ()")))
((last . 1) . (agda2-goals-action '(0 1 2)))
((last . 2) . (agda2-make-case-action '("idTrue' false ()" "idTrue' true refl = ?")))
((last . 1) . (agda2-goals-action '(0 1 2)))
